### PR TITLE
Add Wildcard certificate to ingress resources

### LINKED ISSUE #426: Support wildcard default cert by making TLS secretName not required like k8ix
**Describe the bug**
Specifying a host but not a secret will cause your ingress to be rejected
`ejected: Error retrieving secret for Ingress production-custom-deploy: resource name may not be empty` and the https version to output a 404 error.

This is counter to the best wildcard certificate solution in `kubernetes/ingress-nginx`. Which lets you configure TLS, but leave out the certificate, instead defaulting to your default certificate. In our cases, a wildcard certificate.
https://github.com/kubernetes/ingress-nginx/issues/2170#issuecomment-392855039

**To Reproduce**
1. Configure kubernetes-ingress
2. Configure an ingress with tls, but leave out secretName
3. See error

**Expected behavior**
The default certificate defined in the controller should be used and the rest of the ingress config obayed. This is not easy to workaround as ingress' are often namespaced and wildcard certificates are usually out of reach

**Your environment**
* Version of the Ingress Controller - 1.3.2
* Version of Kubernetes 1.10
* Kubernetes platform GCP
* Using NGINX

### PR DESCRIPTION
**Pull request by Rulox**
### Proposed changes
This PR introduces Wildcard certificates for ingress resources that have TLS termination enabled but don't specify any secretName.

* Closes #426 

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/helm-chart/README.md [hunk 1]
```diff
@@ -69,6 +69,9 @@ Parameter | Description | Default
...
 `controller.defaultTLS.key` | The base64-encoded TLS key for the default HTTPS server. **Note:** If not specified, a pre-generated key is used. It is recommended that you specify your own key. | A pre-generated key.
 `controller.defaultTLS.secret` | The secret with a TLS certificate and key for the default HTTPS server. The value must follow the following format: `<namespace>/<name>`. Used as an alternative to specifiying a certifcate and key using `controller.defaultTLS.cert` and `controller.defaultTLS.key` parameters. | None
+`controller.wildcardTLS.cert` | The base64-encoded TLS certificate for every Ingress host that has TLS enabled but no secret specified. If the parameter is not set, for such Ingress hosts NGINX will break any attempt to establish a TLS connection. | None
+`controller.wildcardTLS.key` | |The base64-encoded TLS key for every Ingress host that has TLS enabled but no secret specified. If the parameter is not set, for such Ingress hosts NGINX will break any attempt to establish a TLS connection. | None
```
Comment (pleshakov) [inline deployments/helm-chart/README.md]:
```suggestion
`controller.wildcardTLS.key` | The base64-encoded TLS key for every Ingress host that has TLS enabled but no secret specified. If the parameter is not set, for such Ingress hosts NGINX will break any attempt to establish a TLS connection. | None
```

File: deployments/helm-chart/templates/controller-daemonset.yaml [hunk 2]
```diff
@@ -82,6 +82,11 @@ spec:
           - -external-service={{ .Values.controller.reportIngressStatus.externalService }}
           - -enable-leader-election={{ .Values.controller.reportIngressStatus.enableLeaderElection }}
 {{- end }}
+{{- if .Values.controller.wildcardTLS.secret }}
+          - -wildcard-tls-secret={{ .Values.controller.wildcardTLS.secret }}
+{{- else if and .Values.controller.wildcardTLS.cert .Values.controller.wildcardTLS.key }}
+          - -wildcard-tls-secret=wildcard-tls-secret
```
Comment (pleshakov) [inline deployments/helm-chart/templates/controller-daemonset.yaml]: ```<namespace>/<name>?```

Comment (pleshakov) [reply deployments/helm-chart/templates/controller-daemonset.yaml]: same in controller-deployment.yaml

Comment (Rulox) [reply deployments/helm-chart/templates/controller-daemonset.yaml]: Good catch! thanks

**Comment by Rulox** [comment]
@pleshakov changed! Please re-review.

**Pull request merged**